addons/project/models/project_task.py

```diff
@@ -1912,35 +1274,45 @@ def action_convert_to_subtask(self):
             }
         }
 
-    def action_convert_to_template(self):
-        self.ensure_one()
-        if not self.project_id:
-            return {
-                'type': 'ir.actions.client',
-                'tag': 'display_notification',
-                'params': {
-                    'type': 'danger',
-                    'message': _('Private tasks cannot be converted into templates'),
-                },
+    def _project_task_template_vals(self):
+        defaults = {}
+        if self.recurring_task:
+            defaults = {
+                'repeat_interval': self.repeat_interval,
+                'repeat_unit': self.repeat_unit,
+                'repeat_type': self.repeat_type,
+                'repeat_until': self.repeat_until,
             }
```

Comment:
Because of the compute method, the default is set during copy operations, so we retrieve it using self